waspc/src/Wasp/Generator/DbGenerator.hs

```diff
@@ -99,7 +110,49 @@ genPrismaSchema wasp = createTemplateFileDraft dstPath tmplSrcPath (Just templat
 
 genMigrationsDir :: Wasp -> Maybe FileDraft
 genMigrationsDir wasp =
-  (getMigrationsDir wasp) >>= \waspMigrationsDir ->
+  getMigrationsDir wasp >>= \waspMigrationsDir ->
     Just $ createCopyDirFileDraft (SP.castDir genProjectMigrationsDir) (SP.castDir waspMigrationsDir)
   where
     genProjectMigrationsDir = dbRootDirInProjectRootDir </> dbMigrationsDirInDbRootDir
+
+writeDbSchemaChecksumToFile :: Path' Abs (Dir ProjectRootDir) -> IO ()
+writeDbSchemaChecksumToFile genProjectRootDir = do
+  dbSchemaExists <- doesFileExist dbSchemaFp
+  when dbSchemaExists $ do
+    checksum <- calculateChecksumStringForFile dbSchemaFp
+    writeFile dbSchemaChecksumFp checksum
+  where
+    dbSchemaFp = SP.fromAbsFile $ genProjectRootDir </> dbSchemaFileInProjectRootDir
+    dbSchemaChecksumFp = SP.fromAbsFile $ genProjectRootDir </> dbSchemaChecksumFileInProjectRootDir
+
+checkIfDbSchemaMatchesChecksum :: Path' Abs (Dir ProjectRootDir) -> ExceptT String IO ()
+checkIfDbSchemaMatchesChecksum genProjectRootDir = do
+  dbSchemaExists <- liftIO $ doesFileExist dbSchemaFp
+  let maybeDbSchemaFp = if dbSchemaExists then Just dbSchemaFp else Nothing
+  dbSchemaChecksumExists <- liftIO $ doesFileExist dbSchemaChecksumFp
+  let maybeDbSchemaChecksumFp = if dbSchemaChecksumExists then Just dbSchemaChecksumFp else Nothing
+  checkIfDbSchemaMatchesChecksum' maybeDbSchemaFp maybeDbSchemaChecksumFp
+  where
+    dbSchemaFp = SP.fromAbsFile $ genProjectRootDir </> dbSchemaFileInProjectRootDir
+    dbSchemaChecksumFp = SP.fromAbsFile $ genProjectRootDir </> dbSchemaChecksumFileInProjectRootDir
+
+checkIfDbSchemaMatchesChecksum' :: Maybe FilePath -> Maybe FilePath -> ExceptT String IO ()
+checkIfDbSchemaMatchesChecksum' (Just dbSchemaFp) (Just dbSchemaChecksumFp) = do
+  dbSchemaFileChecksum <- liftIO $ calculateChecksumStringForFile dbSchemaFp
```

Comment:
I wonder if instead of receving file paths as maybes, it would be better to just catch possible errors here in this function while trying to read the files. That way we don't have to make assumptions about the external function correctly checking if files exist or not, and it also might simplify our logic, not sure?

Reply:
Yeah sure I can give it a go 👍🏻

Reply:
I see what you mean -> I am not sure myself what is nicer, maybe (put intended) give it a try and then we will see what is nicer, as you said.